File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    void levelOrder(TreeNode * root,vector<vector<int>>&ans,int level){
        if(root==NULL)return;
        if(level>=ans.size())ans.push_back({});
        ans[level].push_back(root->val);
        levelOrder(root->left,ans,level+1);
        levelOrder(root->right,ans,level+1);
    }
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>>ans;
        levelOrder(root,ans,0);
        return ans;
    }
};

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
    if (!root) return [];
    const result = [];
    const queue = [root];
    while (queue.length) {
        let size = queue.length;
        let level = [];
        while (size) {
            const currentNode = queue.shift();
            level.push(currentNode.val);
            if (currentNode.left) queue.push(currentNode.left);
            if (currentNode.right) queue.push(currentNode.right);
            size--;
        }
        result.push(level);

    }
    return result;

};